types/V1.mli

```diff
@@ -125,40 +125,38 @@ module type FLOW = sig
-      segment of data from the current flow, receives an [Eof]
-      signifying that the connection is now closed, or an [Error]. *)
+  (** [read flow] reads some data from the flow and returns a
+      fresh buffer (of arbitrary size), an [`Eof] if the other side has
+      called [close] or an [`Error]. *)
 
   val write: flow -> buffer -> [`Ok of unit | `Eof | `Error of error ] io
-  (** [write flow buffer] will block until [buffer] has been added to
-      the send queue. There is no indication when the buffer has
-      actually been read and, therefore, it must not be reused.  The
-      contents may be transmitted in separate packets, depending on
-      the underlying transport. The result [`Ok ()] indicates success,
-      [`Eof] indicates that the connection is now closed and [`Error]
-      indicates some other error. *)
+  (** [write flow buffer] writes a buffer to the flow. There is no
+      indication when the buffer has actually been read and, therefore,
+      it must not be reused.  The contents may be transmitted in
+      separate packets, depending on the underlying transport. The
+      result [`Ok ()] indicates success, [`Eof] indicates that the
+      connection is now closed and [`Error] indicates some other error. *)
 
   val writev: flow -> buffer list -> [`Ok of unit | `Eof | `Error of error ] io
-  (** [writev flow buffers] will block until the buffers have all been
-      added to the send queue. There is no indication when the buffers
-      have actually been read and, therefore, they must not be reused.
-      The result [`Ok ()] indicates success, [`Eof] indicates that the
-      connection is now closed and [`Error] indicates some other
-      error. *)
+  (** [writev flow buffers] writes a sequence of buffers to the flow.
+      There is no indication when the buffers have actually been read and,
+      therefore, they must not be reused. The result [`Ok ()] indicates
+      success, [`Eof] indicates that the connection is now closed and
+      [`Error] indicates some other error. *)
 
   val close: flow -> unit io
-  (** [close flow] will flush all pending writes and signal the remote
+  (** [close flow] flushes all pending writes and signals the remote
       endpoint that there will be no future writes. Once the remote endpoint
       has read all pending data, it is expected that calls to [read] on
-      the remote will return [`Eof].
+      the remote return [`Eof].
 
       Note it is still possible for the remote endpoint to [write] to
       the flow and for the local endpoint to call [read]. This state where
       the local endpoint has called [close] but the remote endpoint
       has not called [close] is similar to that of a half-closed TCP
       connection or a Unix socket after [shutdown(SHUTDOWN_WRITE)].
 
-      The result [unit io] will become determined when the remote endpoint
+      The result [unit io] determines when the remote endpoint
```

Comment:
Yeah I agree -- I think when I originally wrote the word "determined" I was thinking about Async terminology, but we don't use the word anywhere else. I'll try your suggestion.